lib/pure/os.nim

```diff
@@ -1634,6 +1634,24 @@ proc setFilePermissions*(filename: string, permissions: set[FilePermission]) {.
       var res2 = setFileAttributesA(filename, res)
     if res2 == - 1'i32: raiseOSError(osLastError(), $(filename, permissions))
 
+const hasCCopyfile = defined(osx) and not defined(nimLegacyCopyFile)
+  # xxx instead of `nimLegacyCopyFile`, support something like: `when osxVersion >= (10, 5)`
+
+when hasCCopyfile:
+  # `copyfile` API available since osx 10.5.
+  {.push nodecl, header: "<copyfile.h>".}
+  type
+    copyfile_state_t {.nodecl.} = pointer
```

Comment:
why it is special?

Reply:
the other type definition doesn't has nodecl pragma

Reply:
* if i remove nodecl here `nim r --lib:lib -b:cpp tests/stdlib/tos.nim` would fail on OSX (see Example 2 from https://github.com/nim-lang/Nim/issues/16890 which I just added)
* the other type definition is cint and is an exact match in C declaration, whereas `copyfile_state_t` isn't really a `pointer` but a `copyfile_state_t` which in copyfile.h is `_copyfile_state*`; i can't use `_copyfile_state` directly as it's an implementation detail so i must use `copyfile_state_t` directly, which is fine with nodecl since I'm not creating a wrong C declaration for it